src/AppInstallerRepositoryCore/Microsoft/Schema/Checkpoint_1_0/CheckpointArgumentsTable.cpp

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "CheckpointArgumentsTable.h"
+#include "SQLiteStatementBuilder.h"
+#include "Microsoft/Schema/ICheckpointIndex.h"
+
+namespace AppInstaller::Repository::Microsoft::Schema::Checkpoint_V1_0
+{
+    using namespace SQLite;
+    using namespace std::string_view_literals;
+
+    static constexpr std::string_view s_CheckpointArgumentsTable_Table_Name = "CheckpointArguments"sv;
```

Comment:
Instead of having a column for each argument type, wouldn't it make more sense to have an `ArgumentName` table which associates a unique numerical ID to each argument type, then use a many-to-one architecture with [dynamic typing](https://sqlite.org/faq.html#q3) for storing the arguments? It would reduce the amount of space needed in memory to not have to have empty columns for every argument, especially when arguments aren't supported by certain commands

![image](https://github.com/microsoft/winget-cli/assets/12611259/317d2778-78c9-4372-915b-ae4053cd2165)